src/App.tsx

```diff
@@ -1,61 +1,125 @@
+import React, { useState } from 'react';
 import './App.scss';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList/TodoList';
+import { User } from './types/User';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+const getUserById = (id: number): User | undefined => {
+  return usersFromServer.find(user => user.id === id);
+};
+
+const initializedTodos = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
+
+export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>(initializedTodos);
+  const [selectedUser, setSelectedUser] = useState<number>(0);
+  const [newTodoTitle, setNewTodoTitle] = useState<string>('');
+  const [userError, setUserError] = useState<string>('');
+  const [titleError, setTitleError] = useState<string>('');
+
+  const clear = () => {
+    setNewTodoTitle('');
+    setSelectedUser(0);
+  };
+
+  const handleSelect = (element: React.ChangeEvent<HTMLSelectElement>) => {
+    setSelectedUser(+element.target.value);
+    setUserError('');
+  };
+
+  const handleTitle = (element: React.ChangeEvent<HTMLInputElement>) => {
+    setNewTodoTitle(element.target.value);
+    setTitleError('');
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    let hasError = false;
+
+    if (!selectedUser || selectedUser === 0) {
+      setUserError('Please choose a user');
+      hasError = true;
+    } else {
+      setUserError('');
+    }
+
+    if (newTodoTitle === '') {
```

Comment:
```suggestion
    if (newTodoTitle) {
```

can be more simplify